src/Common/logger_useful.h

```diff
@@ -7,6 +7,29 @@
 #include <Poco/Message.h>
 #include <Common/CurrentThread.h>
 
+/// This wrapper is useful to save formatted message into a String before sending it to a logger
+class LogToStrImpl
```

Comment:
Don't have a useful suggestion just wanted to say that this is a really nice idea to avoid double formatting.